AutoCollection/HttpDependencies.ts

```diff
@@ -122,31 +122,40 @@ class AutoCollectHttpDependencies {
         // methods exist before invoking them.
         if (Util.canIncludeCorrelationHeader(client, requestParser.getUrl()) && telemetry.request.getHeader && telemetry.request.setHeader) {
             if (client.config && client.config.correlationId) {
-                const correlationHeader = telemetry.request.getHeader(RequestResponseHeaders.requestContextHeader);
-                if (correlationHeader) {
-                    const components = correlationHeader.split(",");
-                    const key = `${RequestResponseHeaders.requestContextSourceKey}=`;
-                    if (!components.some((value) => value.substring(0,key.length) === key)) {
-                        telemetry.request.setHeader(
-                            RequestResponseHeaders.requestContextHeader,
-                            `${correlationHeader},${RequestResponseHeaders.requestContextSourceKey}=${client.config.correlationId}`);
+                // getHeader returns "any" type in newer versions of node. In basic scenarios, this will be <string | string[] | number>, but could be modified to anything else via middleware
+                const correlationHeader = <any>telemetry.request.getHeader(RequestResponseHeaders.requestContextHeader)
+                let header: string; // cast correlationHeader to string
+                if (typeof correlationHeader === "string") {
+                    header = correlationHeader;
+                } else if (correlationHeader instanceof Array) { // string[]
+                    header = correlationHeader.join(",");
+                } else if (correlationHeader && typeof (correlationHeader as any).toString === "function") {
+                    // best effort attempt: requires well-defined toString
+                    try {
+                        header = (correlationHeader as any).toString();
```

Comment:
Do you know if there's any functional or performance differences between `header.toString()` and `'' + header` or `String(header)`?

Reply:
https://stackoverflow.com/questions/3945202/whats-the-difference-between-stringvalue-vs-value-tostring
It seems `'' + header` and `String(header)` are equivalent, and they attempt to call `toString` (and `valueOf` if that failed). For perf, it seems toString is slightly better, but using either of the other two would cover the scenario of `valueOf` being properly defined instead of `toString` for a custom object.